velox/exec/fuzzer/WriterFuzzer.cpp

```diff
@@ -596,5 +717,66 @@ RowTypePtr WriterFuzzer::generateOutputType(
   return {ROW(std::move(outputNames), std::move(outputTypes))};
 }
 
+std::string WriterFuzzer::sortSql(
+    const std::shared_ptr<HiveConnectorSplit>& split,
+    const std::vector<std::string>& names,
+    const std::vector<TypePtr>& types,
+    int32_t partitionOffset,
+    const std::vector<std::string>& partitionKeys,
+    const std::vector<std::shared_ptr<const HiveSortingColumn>> sortBy) {
+  // For a split, extract the partition filters and bucket filters.
+  std::stringstream whereSql;
+  whereSql << "WHERE ";
+  for (int i = partitionOffset; i < partitionOffset + partitionKeys.size();
+       ++i) {
+    const auto& partitionKey = names.at(i);
+    auto partitionValue = split->partitionKeys.at(partitionKey);
+    if (partitionValue.has_value()) {
+      whereSql << partitionKey << " = "
+               << partitionToSql(types.at(i), partitionValue.value());
+    } else {
+      whereSql << partitionKey << " IS NULL";
+    }
+    whereSql << " AND ";
+  }
+  whereSql << "\"$bucket\" = " << split->tableBucketNumber.value();
+
+  std::stringstream selectedColumns;
+  for (int i = 0; i < sortBy.size(); ++i) {
+    if (i > 0) {
+      selectedColumns << ", ";
+    }
+    selectedColumns << sortBy.at(i)->sortColumn();
+  }
+  return "SELECT " + selectedColumns.str() + " FROM tmp_write " +
+      whereSql.str();
+}
+
+std::string WriterFuzzer::partitionToSql(
+    const TypePtr& type,
+    const std::string& partitionValue) {
+  if (type->isVarchar()) {
+    return "'" + std::regex_replace(partitionValue, std::regex("'"), "''") +
+        "'";
+  }
+  return partitionValue;
+}
+
+void WriterFuzzer::checkEqualVectors(
```

Comment:
Can we use assertEqualResults? in QueryAsserations.h? Thanks!

Reply:
That will compare result without considering order